### PR TITLE
Optimize merging array elements

### PR DESCRIPTION
**Pull request by wgliang**
Optimize merging array elements

### CODE REVIEW

**Comment by winkyao** [comment]
Thanks for contribution, well done.

**Comment by disksing** [comment]
LGTM.

**Comment by winkyao** [comment]
@wgliang Please fix ci, you can run `make dev` locally to check it.

File: util/ranger/new_refiner.go [hunk 1]
```diff
@@ -215,9 +215,7 @@ func BuildRange(sc *variable.StatementContext, conds []expression.Expression, ra
 			return nil, nil, nil, errors.Trace(err)
 		}
 		retRanges = make([]types.Range, 0, len(ranges))
-		for _, ran := range ranges {
-			retRanges = append(retRanges, ran)
-		}
+		retRanges = append(retRanges, ranges...)
```
Comment (winoros) [inline util/ranger/new_refiner.go]: This will make compile failed.

**Pull request merged**